Sources/Orbit/Support/Components/TimelineItem.swift

```diff
@@ -0,0 +1,291 @@
+    }
+}
+
+// MARK: - Types
+
+public enum TimelineItemType: Equatable {
+    case past
+    case current(Status)
+    case future
+
+    public var iconSymbol: Icon.Symbol {
+        switch self {
+            case .past:                  return .checkCircle
+            case .current(.critical):    return .closeCircle
+            case .current(.warning):     return .alertCircle
+            case .current(.success):     return .checkCircle
+            case .current(.info):        return .none
+            case .future:                return .none
+        }
+    }
+
+    public var color: Color {
+        switch self {
+            case .past:                  return Status.success.color
+            case .current(.critical):    return Status.critical.color
+            case .current(.warning):     return Status.warning.color
+            case .current(.success):     return Status.success.color
+            case .current(.info):        return Status.success.color
+            case .future:                return .cloudNormalHover
+        }
+    }
+
+    public var isCurrentStep: Bool {
+        switch self {
+            case .current:
+                return true
+            case .past, .future:
+                return false
+        }
+    }
+
+    public var textColor: UIColor {
+        isCurrentStep ? .inkDark : .inkLight
+    }
+
+    public var isLineDashed: Bool {
+        switch self {
+            case .future:
+                return true
+            case .past, .current:
+                return false
+        }
+    }
+}
+
+struct TimelineItemPreferenceKey: SwiftUI.PreferenceKey {
+
+    typealias Value = [TimelineItemPreference]
+
+    static var defaultValue: Value = []
+
+    static func reduce(value: inout Value, nextValue: () -> Value) {
+        value.append(contentsOf: nextValue())
+    }
+}
+
+struct TimelineItemPreference {
+
+    let bounds: Anchor<CGRect>
+    let type: TimelineItemType
+}
+
+// MARK: - Previews
+struct TimelineItemPreviews: PreviewProvider {
+
+    static var previews: some View {
+        PreviewWrapper {
+            standalone
+            snapshots
+        }
+        .previewLayout(.sizeThatFits)
+    }
+
+    static var standalone: some View {
+        TimelineItem(
+            "Requested",
+            sublabel: "3rd May 14:04",
+            type: .past,
+            description: "We’ve assigned your request to one of our agents."
+        )
+        .padding()
+    }
+
+    static var snapshots: some View {
+        VStack(alignment: .leading, spacing: .xLarge) {
+            TimelineItem(
+                "Requested",
+                sublabel: "3rd May 14:04",
+                type: .future,
+                description: "We’ve assigned your request to one of our agents."
+            )
+            TimelineItem(
+                "Requested",
+                sublabel: "3rd May 14:04",
+                type: .current(.info),
```

Comment:
IMO it is not natural to mark the "current"/"in progress" step with info style - info style looks blue in the whole Orbit. Reusing it here doesn't make user confident what to use.

Reply:
If needed, we can create specific `TimelineIndicator.Status` type that will be limited compared to normal `Status` (the status is mostly a property of the indicator/line anyway)

Reply:
In Figma they use 3 out of 4 Status states and then default and active

<img width="134" alt="image" src="https://user-images.githubusercontent.com/13184609/200300900-7e0753db-a267-420d-a59b-b54612271beb.png">


<img width="582" alt="image" src="https://user-images.githubusercontent.com/13184609/200298764-a160fdd1-c824-40af-bf0a-4ae9cce43807.png">

What would you propose in this case? 
1. Extend the `Type` enum and just ignore `.current(.info)`:
```swift
enum TimelineItemType  {
    case past
    case status(Status)
    case active
    case future
}
```
2. Untie from `Status` completely:
```swift
enum TimelineItemType  {
    case past
    case currentWarning
    case currentSuccess
    case currentCritical
    case currentActive
    case future
}
```

Or something else?

Reply:
If i did not forget some state, maybe it could be:
 

```
extension TimelineIndicator {

   enum Progress {
      case past          // Always 'done'
      case present(Status? = nil)
      case future      // Always 'grey'
  }

   enum Status {
      case warning
      case critical
  }
}
```

Reply:
Updated, although kept it as `TimelineItemType` to maintain the naming as in FIgma